contracts/libraries/ERC721Booking.sol

```diff
@@ -178,10 +209,13 @@ abstract contract ERC721Booking is Context, ERC165, IERC721, IERC721Metadata, Re
 
         _beforeTokenTransfer(from, to, fromId, toId);
 
-        // Underflow of the sender's balance is impossible because we check for
-        // ownership above and the recipient's balance can't realistically overflow.
         uint256 amount = toId - fromId + 1;
-        if (_bookedBy[fromId] != address(0)) {
+        uint256 bookingId = bookingNo + 1;
```

Comment:
I have changed the code to align with the updated booking process, which specifies that a booking is created once Nite tokens are transferred from the zero address. To cancel, the guest needs to return the tokens to the zero address, which will delete the current booking data. This will enable the host to use the tokens for a new booking. What do you think?